### ISSUE TITLE
Update sycl:: functionality deprecated for 2020 specification

### ISSUE THREAD

**Issue by aelizaro** [oneapi-src/oneMKL#110]
# Summary

Align all sycl:: apis with sycl 2020 spec version

# Problem statement
llvm compiler have already added new APIs and deprecated old ones (Example: get_count() for sycl::buffer), but dpcpp compiler doesn't support new versions yet.

# Preferred solution
- Update functionality marked as deprecated in sycl 2020 to the new versions
- Remove SYCL2020_DISABLE_DEPRECATION_WARNINGS macro

**Comment by mmeterel** [comment]
@dnhsieh-intel Can we keep this one open or open a new issue for the backends where #167 is not covering?

**Comment by dnhsieh-intel** [comment]
Definitely. This was closed automatically by github. I guess github grabbed it from the description in #167. I will open a new issue for the problem of cuBLAS, but I prefer to keep this one open since we still have cuSOLVER.

**Comment by dnhsieh-intel** [comment]
Reopen this issue as we still not cover cuBLAS and cuSOLVER.
